src/library/scala/collection/immutable/ChampCommon.scala

```diff
@@ -66,8 +66,8 @@ private[immutable] abstract class ChampBaseIterator[T <: Node[T]] {
   protected var currentValueNode: T = _
 
   private var currentStackLevel: Int = -1
-  private val nodeCursorsAndLengths: Array[Int] = Array.ofDim[Int](MaxDepth * 2)
-  private val nodes: Array[T] = Array.ofDim[Node[T]](MaxDepth).asInstanceOf[Array[T]]
+  private val nodeCursorsAndLengths: Array[Int] = new Array[Int](MaxDepth * 2)
+  private val nodes: Array[T] = new Array[Node[T]](MaxDepth).asInstanceOf[Array[T]]
```

Comment:
Why do we have `Array.ofDim` in the first place? Just for symmetry with the higher-dimension ones?

And why is it not marked `@inline`? This looks like an optimization the user should never have to perform manually.

Reply:
I suppose, it is for symmetry with `List`, `Vector`, etc.

`@inline` doesn't seem to be sufficient to eliminate the overhead.

```
⚡ cat sandbox/test.scala && scalac -opt:l:inline -opt-inline-from:** -opt-warnings -Yopt-log-inline _ sandbox/test.scala && javap -cp . -c Dim
class Dim {
  @inline final def inlinableOfDim[T: reflect.ClassTag](n1: Int): Array[T] = new Array[T](n1)
  def test1 = inlinableOfDim[Int](42)
  def test2 = Array.ofDim[Int](42)
  def test3 = new Array[Int](42)
}
Inline into Dim.test1: inlined Dim.inlinableOfDim. Before: 11 ins, inlined: 8 ins.
Compiled from "test.scala"
public class Dim {
  public final <T> java.lang.Object inlinableOfDim(int, scala.reflect.ClassTag<T>);
    Code:
       0: aload_2
       1: iload_1
       2: invokeinterface #18,  2           // InterfaceMethod scala/reflect/ClassTag.newArray:(I)Ljava/lang/Object;
       7: areturn

  public int[] test1();
    Code:
       0: bipush        42
       2: getstatic     #30                 // Field scala/reflect/ClassTag$.MODULE$:Lscala/reflect/ClassTag$;
       5: invokevirtual #34                 // Method scala/reflect/ClassTag$.Int:()Lscala/reflect/ClassTag;
       8: astore_2
       9: istore_1
      10: aload_2
      11: iload_1
      12: invokeinterface #18,  2           // InterfaceMethod scala/reflect/ClassTag.newArray:(I)Ljava/lang/Object;
      17: checkcast     #36                 // class "[I"
      20: areturn

  public int[] test2();
    Code:
       0: getstatic     #44                 // Field scala/Array$.MODULE$:Lscala/Array$;
       3: bipush        42
       5: getstatic     #30                 // Field scala/reflect/ClassTag$.MODULE$:Lscala/reflect/ClassTag$;
       8: invokevirtual #34                 // Method scala/reflect/ClassTag$.Int:()Lscala/reflect/ClassTag;
      11: invokevirtual #47                 // Method scala/Array$.ofDim:(ILscala/reflect/ClassTag;)Ljava/lang/Object;
      14: checkcast     #36                 // class "[I"
      17: areturn

  public int[] test3();
    Code:
       0: bipush        42
       2: newarray       int
       4: areturn

  public Dim();
    Code:
       0: aload_0
       1: invokespecial #52                 // Method java/lang/Object."<init>":()V
       4: return
}
```

We'd also have to mark `ClassTag#newArray` as `@inlinable`, but even then the inliner wouldn't have enough knowledge to collapse that into `newarray int` because.

It would be easier to optimize this a bit early, where we still have types around. We do some other Array transformations in the `Cleanup` phase already, maybe that's the spot.